snapshots/testsuite/testsuite.go

```diff
@@ -785,6 +806,10 @@ func checkRemove(ctx context.Context, t *testing.T, snapshotter snapshots.Snapsh
 // checkSnapshotterViewReadonly ensures a KindView snapshot to be mounted as a read-only filesystem.
 // This function is called only when WithTestViewReadonly is true.
 func checkSnapshotterViewReadonly(ctx context.Context, t *testing.T, snapshotter snapshots.Snapshotter, work string) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Read-only protection on views is not supported on WCOW")
+	}
+
```

Comment:
This test disagrees with [the API docs](https://github.com/containerd/containerd/blob/58701f87c7db5abf5c203a4a8d8b8558977d06cf/snapshots/snapshotter.go#L301-L302), as the latter states that the mount from a View _may_ be read-only, and that writes to the mount must be _ignored_, not that they must fail).

I think what it should do is if the View result is not read-only (i.e. if the write doesn't fail), request the View again and ensure that the file written does not exist in the second View.